### PR TITLE
UnixPB: Include Pre-Req Packages For SSL Test

### LINKED ISSUE #3059: Dependencies for ssl-tests testsuite
Integration of new ssl-tests testsuite into aqa is in progress [1]. However some additional dependencies will be needed.

For linux these are (ubuntu package names):
- `make` `openssl`
-  `curl`
- `gnutls-bin` (needed by ssl-tests-gnutls-client.sh)
- `libnss3` `libnss3-tools` `libnss3-dev` `gcc` `pkg-config` (needed by ssl-tests-nss-client.sh)

For other systems (Windows and Mac, possibly others?):
- `make` `openssl`
- `curl`
- integration testing with other clients currently not enable on non-linux

For more details, see [1].

[1] https://github.com/adoptium/aqa-tests/issues/4403

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #3059 

Install packages required for running ssl-tests for 3rd party clients ( gnutls / nss / openssl ), also include appropriate documentation update, and an additional test has been added to the VPC tests.

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] other documentation is changed or added (if applicable)
- [x] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)

### CODE REVIEW

**Comment by steelhead31** [comment]
VPC run with SSL tests on Fed35.. [https://ci.adoptium.net/job/VagrantPlaybookCheck/OS=Fedora35,label=vagrant/1703/console](https://ci.adoptium.net/job/VagrantPlaybookCheck/OS=Fedora35,label=vagrant/1703/console)  
  
VPC Assorted other platforms  
https://ci.adoptium.net/job/VagrantPlaybookCheck/1704/

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/openSUSE.yml [hunk 1]
```diff
@@ -62,4 +65,7 @@ Test_Tool_Packages:
   - pulseaudio
   - xorg-x11
   - xorg-x11-devel
-  - glibc-locale                  # Internationalization tests
+  - glibc-locale
+  - gnutls
+  - libnss3.so
+  - mozilla-nss                 # Internationalization tests
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/openSUSE.yml:71]: Should this not all be aligned with the locale package install?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml [hunk 2]
```diff
@@ -25,13 +25,13 @@
 
 - name: Add AdoptOpenJDK GPG key
   apt_key:
-    url: https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public
+    url: https://packages.adoptium.net/artifactory/api/gpg/key/public
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml:28]: This should be a separate PR

**Review by sxa — COMMENTED** [review]
Two things
1. I presume this doesn't modify any existing openssl install or links etc  that might affect anything building against it (e.g. openh9)
2. Are we happy that this will be ok on CentOS6?
3. This will presumably need to be added to the Dockerfiles in the DockerStatic role since those are used for most of our test machines

**Review by sxa — CHANGES_REQUESTED** [review]
Requesting changes as this will need to be in the docker files

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml (hunk 2 above)
Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml]: This has been reverted, and a seperate PR is on its way.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/openSUSE.yml (hunk 1 above)
Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/openSUSE.yml:71]: 1.  This definitely doesnt impact any existing openssl install, its additional tools to allow the tests to run, builds and tests have been run via VPC
2.  I haven't specifically tested CentOS6 ( I have 7 ), I will do some additional testing on this
3.  I'll get the dockerstatic files updated appropriately, can possibly look to roll them out via the docker package update process.

Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/openSUSE.yml:71]: The packages install and the playbooks run OK on CentOS6, and the tests execute OK, the ./ssl-tests-openssl-client.sh tests run OK , but there are a couple of issues in the other two test suites ( which I've sent over to zzambers ) these are 

```
ssl-tests-gnutls-client.sh

IGNORED: SunJSSE/DTLSv1.2: DTLSv1.0 + TLS_EMPTY_RENEGOTIATION_INFO_SCSV
Exception in thread "main" java.lang.RuntimeException: java.lang.IllegalArgumentException: Unknown protocol: CTYPE-OPENPGP
	at SSLSocketTester.testConfigurations(SSLSocketTester.java:283)
	at SSLSocketTester.testProvider(SSLSocketTester.java:234)
	at SSLSocketTester.testProviders(SSLSocketTester.java:190)
	at Main.main(Main.java:30)
Caused by: java.lang.IllegalArgumentException: Unknown protocol: CTYPE-OPENPGP
	at GnutlsClient.getJavaProtoName(GnutlsClient.java:88)
	at GnutlsClient.getSupportedCiphers(GnutlsClient.java:125)
	at SSLSocketTester.testConfigurations(SSLSocketTester.java:268)
	... 3 more
../ssl-tests/ssl-tests.mk:48: recipe for target 'ssl-tests-run' failed

```

And 

```
ssl-tests-nss-client.sh

../ssl-tests/ssl-tests.mk:48: recipe for target 'ssl-tests-run' failed
make: *** [ssl-tests-run] Error 1

FAILED: SunJSSE/TLSv1.3: TLSv1.3 + TLS_AES_256_GCM_SHA384
Sep 27, 2023 9:53:58 AM SSLSocketServer$1 run
SEVERE: null
javax.net.ssl.SSLHandshakeException: Client requested protocol TLSv1.2 is not enabled or supported in server context
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:117)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:365)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:321)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:312)
	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.negotiateProtocol(ClientHello.java:871)
	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.onClientHello(ClientHello.java:823)
	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.consume(ClientHello.java:801)
	at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:396)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:480)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:458)
	at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:201)
	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1510)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1425)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455)
	at java.base/sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:925)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1016)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:976)
	at SSLSocketServer.serverLoop(SSLSocketServer.java:133)
	at SSLSocketServer$1.run(SSLSocketServer.java:75)
	at java.base/java.lang.Thread.run(Thread.java:833)

stderr: tstclnt: read from socket failed: SSL_ERROR_PROTOCOL_VERSION_ALERT: Peer reports incompatible or unsupported protocol version.
Sep 27, 2023 9:53:58 AM SSLSocketTester testConfiguration
SEVERE: null
java.lang.RuntimeException: Program exit value not zero: 1
	at ExternalClient.test(ExternalClient.java:267)
	at SSLSocketTester.testConfiguration(SSLSocketTester.java:392)
	at SSLSocketTester.testConfigurations(SSLSocketTester.java:322)
	at SSLSocketTester.testProvider(SSLSocketTester.java:234)
	at SSLSocketTester.testProviders(SSLSocketTester.java:190)
	at Main.main(Main.java:30)

FAILED: SunJSSE/TLSv1.3: TLSv1.3 + TLS_AES_128_GCM_SHA256
Sep 27, 2023 9:53:58 AM SSLSocketServer$1 run
SEVERE: null
javax.net.ssl.SSLHandshakeException: Client requested protocol TLSv1.2 is not enabled or supported in server context
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:117)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:365)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:321)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:312)
	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.negotiateProtocol(ClientHello.java:871)
	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.onClientHello(ClientHello.java:823)
	at java.base/sun.security.ssl.ClientHello$ClientHelloConsumer.consume(ClientHello.java:801)
	at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:396)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:480)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:458)
	at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:201)
	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1510)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1425)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455)
	at java.base/sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:925)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1016)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:976)
	at SSLSocketServer.serverLoop(SSLSocketServer.java:133)
	at SSLSocketServer$1.run(SSLSocketServer.java:75)
	at java.base/java.lang.Thread.run(Thread.java:833)

stderr: tstclnt: read from socket failed: SSL_ERROR_PROTOCOL_VERSION_ALERT: Peer reports incompatible or unsupported protocol version.
Sep 27, 2023 9:53:58 AM SSLSocketTester testConfiguration
SEVERE: null
java.lang.RuntimeException: Program exit value not zero: 1
	at ExternalClient.test(ExternalClient.java:267)
	at SSLSocketTester.testConfiguration(SSLSocketTester.java:392)
	at SSLSocketTester.testConfigurations(SSLSocketTester.java:322)
	at SSLSocketTester.testProvider(SSLSocketTester.java:234)
	at SSLSocketTester.testProviders(SSLSocketTester.java:190)
	at Main.main(Main.java:30)

```

File: FAQ.md [hunk 3]
```diff
@@ -211,6 +211,62 @@ should have an underscore `_` prepended to it.
 (For the last one, that makes use of the system.custom target added via
 [this PR](https://github.com/AdoptOpenJDK/openjdk-tests/pull/2234))
 
+## Running The SSL Test Suites
```
Comment (sxa) [inline FAQ.md:214]: Picky comment, but given that this doesn't follow the normal pattern for test execution (yet) can we make this a twisty section for now so as not to take up too much extra space in the document, particularly with all the output that's being shown?
```suggestion
<details>
  <summary>Running the SSL test suites</summary>
```
(With a corresponding `</details`> at the end of the section)

This has made me think that it might be useful to have an example for the external tests which are run in containers too.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Alpine.yml [hunk 4]
```diff
@@ -20,6 +20,9 @@ Build_Tool_Packages:
   - freetype-dev
   - grep
   - gnupg
+  - gnutls
+  - gnutls-dev
+  - gnutls-utils
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Alpine.yml:25]:
```suggestion
  - gnutls                        # OpenSSL tests
  - gnutls-dev                    # OpenSSL tests
  - gnutls-utils                  # OpenSSL tests
```
Making this suggestion in the UI with a proportional font but I think I've got the right amount of spaces here. It's good to have a note of why we've added things if they're for a specific purpose.

File: FAQ.md (hunk 3 above)
Comment (steelhead31) [reply FAQ.md:214]: Added a twisty section for the ssl-tests.

Comment (sxa) [reply FAQ.md:214]: Thank you!

Comment (steelhead31) [reply FAQ.md:214]: I'm working on the Dockerstatic Changes, so I'll add an additional twisty section to the FAQ, on running some of tests on a newly created container too.

Comment (sxa) [reply FAQ.md:214]: Is there a reason why we'd need additional documentation for that over and above what's already in the twisty? It shouldn't make a difference what environment you're running it should it?

Comment (steelhead31) [reply FAQ.md:214]: > Is there a reason why we'd need additional documentation for that over and above what's already in the twisty? It shouldn't make a difference what environment you're running it should it?

Having done some of this , no it doesn't :), I don think there's a need for any more doc updates., I'll resolve this once all the dockerstatic updates are done.

Comment (steelhead31) [reply FAQ.md:214]: Dockerstatic files updated , and tested.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/openSUSE.yml (hunk 1 above)
Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/openSUSE.yml]: Fixed the errant comment

**Review by sxa — COMMENTED** [review]
Is there a reason why the Alpine dockerfiles have had the new packages added to the existing `apk` line but for the other distributions they use a separate apt/yum command? Would be nice to be consistent across the distributions for maintenance purposes.

**Comment by steelhead31** [comment]
Ubuntu 21.04 check failure due to "hirsute" repos being removed.

**Review by karianna — APPROVED** [review]

**Comment by steelhead31** [comment]
@sxa I've also run the openj9 crypto tests ( in fact the whole java_security suite ) with a build of openj9 (jdk17), and thats all worked fine in both and Ubuntu/x64 vagrant & VM environments.

**Review by sxa — APPROVED** [review]
Thanks :-)
